File: SubscriptionCleaner.cs
using System.Collections.Generic;

namespace Rose
{
    public class SubscriptionCleaner
    {
        private readonly List<Subscription> subscriptions = new List<Subscription>();

        public void Add(Subscription subscription)
        {
            subscriptions.Add(subscription);
        }

        public void Remove(Subscription subscription)
        {
            subscriptions.Remove(subscription);
        }

        public void Unsubscribe()
        {
            subscriptions.ForEach(s => s.Unsubscribe());
        }
    }
}

File: java/com/kkucharski/rose/SubscriptionCleaner.java
package com.kkucharski.rose;

import java.util.LinkedList;
import java.util.List;

public class SubscriptionCleaner {
    private final List<Subscription> subscriptions = new LinkedList<>();

    public void add(Subscription subscription) {
        subscriptions.add(subscription);
    }

    public void remove(Subscription subscription) {
        subscriptions.remove(subscription);
    }

    public void unsubscribe() {
        subscriptions.forEach(Subscription::unsubscribe);
    }
}
